tsweb/request_id.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package tsweb
+
+import (
+	"context"
+	"net/http"
+)
+
+// RequestID is an opaque identifier for a HTTP request, used to correlate
+// user-visible errors with backend server logs. The RequestID is typically
+// threaded through an HTTP Middleware (WithRequestID) and then can be
+// extracted by HTTP Handlers to include in their logs.
+//
+// RequestID is an opaque identifier for a HTTP request, used to correlate
+// user-visible errors with backend server logs. If present in the context, the
+// RequestID will be printed alongside the message text and logged in the
+// AccessLogRecord.
+//
+// There is no particular format required for a RequestID, but ideally it should
```

Comment:
Could we maybe use a UUID? E.g., `fmt.Sprintf("REQ-%v", uuid.New())` or words to that effect? I'd argue that since the point of `tsweb` is to be opinionated, I don't see much reason to leave the format up to the caller—we could even provide a constructor:

```go
type RequestID string

func NewRequestID() RequestID {
   return fmt.Sprintf("REQ-%v", uuid.New())
}
```

and then the user doesn't need to think about it at all.

Reply:
I did most of the updates above, but shied away from adding a new constructor to avoid new APIs until we need one. But please do let me know if you'd like to include it!

Reply:
There's no longer any way for a caller to set anything _else_ as the request ID, so we can probably simplify this a bit, e.g. maybe:

```go
// A RequestID has the format "REQ-{ID}", and the ID should be treated
// as an opaque string. The current implementation uses a UUID.
```

Reply:
can we instead use `REQ-${base58-of-id}` or similar, I'd love for them to be shorter, the GUID format isn't particularly human-useful, and but it's annoyingly long.